File: homework/Van_Le/homework7.2/SymmetricTree.cs
public class SymmetricTree {
    public bool IsSymmetric(TreeNode root) {
        return IsTheSame(root, root);
    }
    
    private bool IsTheSame(TreeNode tree1, TreeNode tree2){
        if(tree1 == null && tree2 == null) return true;
        if(tree1 == null || tree2 == null) return false;
        
        return tree1.val == tree2.val && IsTheSame(tree1.right, tree2.left)
            && IsTheSame(tree1.left, tree2.right);
    }
}

File: homework/ba-nguyen/lesson-7.2/symmetric-tree.ts
function isSymmetric(root: TreeNode | null): boolean {
  if (!root) return true;
  return isSymmetricRecursive(root.left, root.right);
}

function isSymmetricRecursive(l: TreeNode | null, r: TreeNode | null): boolean {
  if (!l && !r) return true;
  if ((l?.left && !r?.right) || (!l?.right && r?.left)) return false;

  return (
    l?.val === r?.val &&
    isSymmetricRecursive(l.left, r.right) &&
    isSymmetricRecursive(l.right, r.left)
  );
}
